### ISSUE TITLE
REST API profiling in 3.1.0

### ISSUE THREAD

**Issue by bernardotavares** [scottbedard/oc-debugbar-plugin#41]
Hello,

Just upgraded the debugbar from 2.0.0 to 3.1.0, it's working smoothly except I am not able to check API requests

Before I could just make the API call with Postman, go to the Folder icon in the debugbar and choose the latest request.
It worked out of box. Now those requests do not appear.

Am I missing something? Or is this something with barryvdh/laravel-debugbar ~3.0?

**rainlab/debugbar-plugin:** 3.1.0
**octobercms:** build 471
**laravel:** v6.20.13
**php:** 7.2.25

**some configs:**
'debug' => true
'allow_public_access' => true

Thanks

**Comment by LukeTowers** [comment]
See https://github.com/rainlab/debugbar-plugin/blob/master/config/config.php#L27-L29, in order to view those stored requests you need to be authenticated as a backend user with the `rainlab.debugbar.access_stored_requests`

**Comment by bernardotavares** [comment]
Ok, I was thinking it would work like this
- Authenticated in backend with the rainlab.debugbar.access_stored_requests permission or super admin
- Make the request in Postman
- Open the debugbar folder icon and see the stored API request

But it didn't worked for me altough it was not entering this condition in 
plugins\rainlab\debugbar\middleware\InjectDebugbar.php line 40.
```
if (!$user || !$user->hasAccess('rainlab.debugbar.access_stored_requests')) {
  // Disable stored requests
  // Note: this will completely disable storing requests from any users
  // without the required permission. If that functionality is desired again
  // in the future then we can look at overriding the OpenHandler controller
  $this->debugbar->setStorage(null);
}
```
But it turns out that the API request itself was. Just commenting this out solved it for me.
I am only using the debugbar locally so I am not too worried about security.

But is this the expected behavior?

**Comment by LukeTowers** [comment]
The expected behaviour is that only requests from users with that permission will be logged and only those users will be able to view those requests. We could potentially look at adding a separate config value to log all of them though, that sounds like it would probably be the best option for your use case. Thoughts?

**Comment by bernardotavares** [comment]
Honestly I am not sure I understand why we cannot use the allow_public_access for that.
Woudn't people only use it in an environement where they are not concerned about security?

Anyway, if that is the way to go I will submit a PR, I think is important we can use the debugbar with Postman locally without having to worry about this :)

**Comment by LukeTowers** [comment]
At this point we'd have to go with a second config option regardless because people could already have this plugin deployed with that setting enabled and to have it suddenly start logging and displaying logged requests from an update would be terrible from a security perspective. It's an incredibly dangerous feature to enable because it allows anyone with the ability to see the requests to get full admin access by taking over your admin session.

**Comment by bennothommo** [comment]
Fixed by #43.
